apis/path-finder/api/v0/tokens.ts

```diff
@@ -35,22 +32,42 @@ export const DEFAULT_TOKENS_MAP = ZENLINK_DEFAULT_TOKEN_LIST
     return map
   }, new Map())
 
-const { provider } = configureChains([...otherChains] as Chain[], [publicProvider({ priority: 0 })])
-createClient({ provider, autoConnect: true })
-
 async function fetchTokenInfo(chainId: ParachainId, tokenId: string): Promise<Token | undefined> {
   try {
-    const token = await fetchToken({
-      address: tokenId as `0x${string}`,
-      chainId: chainsParachainIdToChainId[chainId],
-    })
-    return new Token({
-      chainId,
-      decimals: token.decimals,
-      name: token.name,
-      symbol: token.symbol,
-      address: token.address,
-    })
+    const client = getClient(chainId)
+    if (!client)
+      return undefined
+
+    const results = await client
+      .multicall({
+        multicallAddress: client.chain?.contracts?.multicall3?.address as Address,
```

Comment:
Latest version of viem added `multicallAddress` inference from the client chain so you don't have to explicitly define it :)